src/lib.rs

```diff
@@ -490,3 +494,54 @@ fn generate_body(trait_: &DeriveTrait, item: &Item) -> TokenStream {
 		}
 	}
 }
+
+/// Removes the `derive_where` attributes from all fields, and variants.
+///
+/// This is needed, because rust does not support those for proc_macro_attribute
+/// currently
+fn input_without_derive_where_attributes(mut input: DeriveInput) -> TokenStream {
+	use syn::Data;
+	let DeriveInput { data, attrs, .. } = &mut input;
+
+	/// Remove all attributes that are `derive_where`
+	fn remove_derive_where(attrs: &mut Vec<Attribute>) {
+		// TODO find the actual path
+		let ident = Ident::new("derive_where", Span::call_site());
+		attrs.retain(|attr| !attr.path.is_ident(&ident))
+	}
+
+	/// Remove all attributes that are `derive_where` from `FieldsNamed`
+	fn remove_derive_where_from_fields_named(fields: &mut FieldsNamed) {
+		let FieldsNamed { named, .. } = fields;
+		named
+			.iter_mut()
+			.for_each(|field| remove_derive_where(&mut field.attrs))
+	}
+
+	/// Remove all attributes that are `derive_where` from `Fields`
+	fn remove_derive_where_from_fields(fields: &mut Fields) {
+		match fields {
+			Fields::Named(fields) => remove_derive_where_from_fields_named(fields),
+			Fields::Unnamed(FieldsUnnamed { unnamed, .. }) => unnamed
+				.iter_mut()
+				.for_each(|field| remove_derive_where(&mut field.attrs)),
+			Fields::Unit => (),
+		}
+	}
+
+	remove_derive_where(attrs);
+	match data {
+		Data::Struct(DataStruct { fields, .. }) => remove_derive_where_from_fields(fields),
+		Data::Enum(DataEnum { variants, .. }) => {
+			variants
+				.iter_mut()
+				.for_each(|Variant { attrs, fields, .. }| {
+					remove_derive_where(attrs);
+					remove_derive_where_from_fields(fields)
+				})
+		}
+		Data::Union(DataUnion { fields, .. }) => remove_derive_where_from_fields_named(fields),
+	}
+
+	input.to_token_stream()
+}
```

Comment:
Should this move in a separate module?

Reply:
I vote for keeping it here. I'm fine either way though.